src/settings/tabconnection.cpp

```diff
@@ -0,0 +1,113 @@
+#include "tabconnection.h"
+
+#include <QVBoxLayout>
+#include <QLineEdit>
+#include <QSettings>
+#include <QLabel>
+
+TabConnection::TabConnection(QWidget *parent)
+    : QWidget(parent)
+{
+    QVBoxLayout *mainLayout = new QVBoxLayout();
+    mainLayout->addStretch(mBasicStretch);
+
+    // getting data from QSettings
+    QSettings s;
+    mUDPportCurrent = (s.contains("defaultUdpPortNumber") ? s.value("defaultUdpPortNumber").toString() : mUDPportDefault);
+    mTCPportCurrent = (s.contains("defaultTcpPortNumber") ? s.value("defaultTcpPortNumber").toString() : mTCPportDefault);
+    mUDPportNew = mUDPportCurrent;
+    mTCPportNew = mTCPportCurrent;
+    mUserName = (s.contains("userName") ? s.value("userName").toString() : "");
+
+    // user name
+    QLabel *userLabel = new QLabel("User name", this);
+    userLabel->setMinimumWidth(mLabelPortNumberWidth);
+    userLabel->setMaximumWidth(mLabelPortNumberWidth);
+    QHBoxLayout *userLayout = new QHBoxLayout;
```

Comment:
Try to avoid fixed widgets size
If there are no other options: it is better to use 1 method setFixedSize()